src/views/Fairdrop/Steps/index.tsx

```diff
@@ -0,0 +1,229 @@
+import { useState } from "react";
+import classNames from "classnames";
+import { useAccount } from "wagmi";
+
+import { fetchWrapper } from "~/utils";
+
+import styles from "./styles.module.sass";
+
+type StepsProps = {
+  next: (val: any) => void;
+};
+
+const Steps: React.FC<StepsProps> = ({ next }) => {
+  const [step, setStep] = useState(0);
+  const [checked, setChecked] = useState(false);
+  const [loading, setLoading] = useState(false);
+  const { data: accountData } = useAccount();
+  const [claimData, setClaimData] = useState({
+    account: String,
+    nonce: String,
+    exipredAt: String,
+    signerSig: String,
+  });
+  const [twitterUrl, setTwitterUrl] = useState('');
+
+  const account = accountData?.address;
+
+  const verifyETHAddress = async () => {
+    setLoading(true);
+    try {
+      const data = await fetchWrapper.get(
+        "/api/fairdrop/nonce?account=" + account
+      );
+      setClaimData(data);
+      setStep(2);
+    } catch (e) {
+      // 已申請過
+      // next("not_eligible");
+      // next("under_review");
+      
+      // 無法申請
+      // next("have_send");
+    }
+    setLoading(false);
+  };
+  const verifyTwitterAccount = () => {
+    window.open(
+      "https://twitter.com/intent/tweet?text=%F0%9F%92%B0Claim%20your%20%24SPACE%F0%9F%92%B0%20at%20https%3A%2F%2Fthespace.game%2Fclaim%20Inspired%20by%20%23RedditPlace%2C%20The%20Space%20is%20the%20world%26%2339%3Bs%20%23NFT%20%23pixelart%20graffiti%20wall%20where%20players%20can%20own%2C%20color%2C%20and%20trade%20pixels%20under%20Harberger%20Tax%20and%20Universal%20Basic%20Income%20%28UBI%29.%20%23TheSpaceGame%20%23%E7%83%8F%E5%A1%97%E9%82%A6",
+      "mywin",
+      "left=0,top=0,width=650,height=650"
+    );
+    setStep(3);
+  };
+  const followUs = () => {
+    window.open(
+      "https://twitter.com/intent/follow?original_referer=https%3A%2F%2Fpublish.twitter.com%2F&ref_src=twsrc%5Etfw%7Ctwcamp%5Ebuttonembed%7Ctwterm%5Efollow%7Ctwgr%5Ethespace2022&screen_name=thespace2022",
+      "mywin",
+      "left=0,top=0,width=650,height=650"
+    );
+    setStep(4);
+  };
+  const claimSpace = async () => {
+    setLoading(true);
+    try {
+      const message =
+        "I am signing this message to prove the ownership of this address.\n\nNonce: " +
```

Comment:
pls reuse https://github.com/thematters/thespace-landing-page/blob/main/src/utils/fairdrop.ts#L1= since API also using it.